### PR TITLE
fix: DEV-1694: Make update_tasks_states async

### PR DESCRIPTION
**Pull request by KonstantinKorotaev**

### CODE REVIEW

File: label_studio/data_manager/actions/basic.py [hunk 1]
```diff
@@ -54,8 +53,13 @@ def delete_tasks(project, queryset, **kwargs):
     # delete only specific tasks
     else:
         # this signal re-save the task back
-        with temporary_disconnect_signal(signals.post_delete, update_is_labeled_after_removing_annotation, Annotation):
+        with temporary_disconnect_all_signals():
             queryset.delete()
+        project.update_tasks_states(
```
Comment (niklub) [inline label_studio/data_manager/actions/basic.py:58]: that will not recompute project.summary counters

File: label_studio/data_manager/actions/basic.py [hunk 2]
```diff
@@ -54,8 +53,13 @@ def delete_tasks(project, queryset, **kwargs):
     # delete only specific tasks
     else:
```
Comment (niklub) [inline label_studio/data_manager/actions/basic.py:54]: we don't need to keep if..else branch if behaviour is  identical

**Review by niklub — CHANGES_REQUESTED** [review]

File: label_studio/data_manager/actions/basic.py [hunk 3]
```diff
@@ -54,8 +53,13 @@ def delete_tasks(project, queryset, **kwargs):
...
         # this signal re-save the task back
-        with temporary_disconnect_signal(signals.post_delete, update_is_labeled_after_removing_annotation, Annotation):
+        with temporary_disconnect_all_signals():
```
Comment (makseq) [inline label_studio/data_manager/actions/basic.py:56]: I'm not sure we can disable all signals here, because we storages use signals to remove annotations

**Review by makseq — APPROVED** [review]

File: label_studio/data_manager/actions/basic.py [hunk 4]
```diff
@@ -43,20 +43,22 @@ def delete_tasks(project, queryset, **kwargs):
...
-            overlap_cohort_percentage_changed=False,
-            tasks_number_changed=True
-        )
 
     # delete only specific tasks
     else:
-        # this signal re-save the task back
-        with temporary_disconnect_signal(signals.post_delete, update_is_labeled_after_removing_annotation, Annotation):
+        # update project summary
+        project.summary.remove_created_annotations_and_labels(Annotation.objects.filter(task__in=queryset))
```
Comment (niklub) [inline label_studio/data_manager/actions/basic.py:51]: could we move all summary recalculation into (a)sync job?

**Review by niklub — CHANGES_REQUESTED** [review]

File: label_studio/data_manager/actions/basic.py (hunk 1 above)
Comment (KonstantinKorotaev) [reply label_studio/data_manager/actions/basic.py]: Fixed

File: label_studio/data_manager/actions/basic.py (hunk 3 above)
Comment (KonstantinKorotaev) [reply label_studio/data_manager/actions/basic.py]: Fixed

**Review by niklub — CHANGES_REQUESTED** [review]

File: label_studio/core/utils/common.py [hunk 5]
```diff
@@ -605,3 +605,39 @@ def round_floats(o):
     if isinstance(o, (list, tuple)):
         return [round_floats(x) for x in o]
     return o
+
+
+class temporary_disconnect_list_signal:
+    """ Temporarily disconnect a list of signals
+        Each signal tuple: (signal_type, signal_method, object)
+        Example:
+            with temporary_disconnect_all_signals(
```
[the remark this answers is not in the document]
Comment (KonstantinKorotaev) [reply label_studio/core/utils/common.py]: Fixed

**Pull request merged**